xmppserver/src/main/java/org/jivesoftware/openfire/handler/IQVersionHandler.java

```diff
@@ -76,6 +83,34 @@ public IQ handleIQ(IQ packet) throws PacketException {
             IQ result = IQ.createResultIQ(packet);
             result.setError(PacketError.Condition.not_acceptable);
             return result;
+        } else if (IQ.Type.result == packet.getType()) {
+            /* handle results coming through BOSH Connections,
+             * other results are processed in org.jivesoftware.openfire.net.SocketRead.java - getIQ()
+             */
+            LocalSession localSession = (LocalSession) XMPPServer.getInstance().getSessionManager().getSession(packet.getFrom());
+
+            Element query = packet.getChildElement();
+            List<Element> elements = query.elements();
+            if (elements.size() > 0) {
+                for (Element element : elements) {
+                    if (element.getName() != null && element.getStringValue() != null) {
+                        if (localSession!=null)
+                        {
+                            localSession.setSoftwareVersionData(element.getName(), element.getStringValue());
+                        }
+                        else
+                        {
+                            /*
+                              The result comes from a server 2 server connection, so we write the information
+                              only to the debug log, because we dont need it at this point.
+                            */
+                            Log.debug(packet.getFrom()+" "+element.getName()+"="+element.getStringValue());
```

Comment:
How expensive is this string concatenation? Would placeholders be better so to avoid doing this each time this code is hit, even when not in debug mode?

Reply:
I changed the logging to Log.debug("XEP-0092 Packet from={} {}={}",packet.getFrom(),element.getName(),element.getStringValue());